### PR TITLE
[nuget] update launch scripts to verify native profiler path

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Avoid crashes when an app using autoinstrumentation NuGet package is built without specifying runtime identifier and launched using launch script.

Fixes #

## What

Verify existence of native profiler at expected location before attempting to launch the app.
OS type/architecture detection in shell script based on similar scripts from this repository (e.g [this one](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/instrument.sh)).

## Tests

CI.
Local smoke tests.

## Checklist

~~- [ ] `CHANGELOG.md` is updated.~~
~~- [ ] Documentation is updated.~~
~~- [ ] New features are covered by tests.~~

### CODE REVIEW

File: nuget/OpenTelemetry.AutoInstrumentation/contentFiles/any/any/instrument.cmd [hunk 1]
```diff
@@ -1,13 +1,42 @@
...
+set COR_PROFILER_PATH=%BASE_PATH%OpenTelemetry.AutoInstrumentation.Native.dll
+
+:: Validate
+IF EXIST %COR_PROFILER_PATH% (
+    set CORECLR_PROFILER_PATH=!COR_PROFILER_PATH!
+) ELSE (
+    set "COR_PROFILER_PATH="
+    echo RuntimeIdentifier was not specified when building the project. Attempting to use the fallback location.
+
+    set COR_PROFILER_PATH_64=%BASE_PATH%runtimes\win-x64\native\OpenTelemetry.AutoInstrumentation.Native.dll
```
Comment (pjanotti) [inline nuget/OpenTelemetry.AutoInstrumentation/contentFiles/any/any/instrument.cmd:15]: Nice, this should work for `dotnet build`.

**Review by pjanotti — APPROVED** [review]
LGTM

**Comment by pjanotti** [comment]
@lachmatt we test the `dotnet` CLI steps for NuGet package, we could have the build mode that caused the crash added to the test - fine to do as a separate PR.

File: nuget/OpenTelemetry.AutoInstrumentation/contentFiles/any/any/instrument.cmd (hunk 1 above)
Comment (RassK) [reply nuget/OpenTelemetry.AutoInstrumentation/contentFiles/any/any/instrument.cmd:15]: maybe "Attempting to use Windows x64 for fallback" to give more information, what is the fallback.

**Review by Kielek — APPROVED** [review]

**Comment by lachmatt** [comment]
There was a [failure](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/actions/runs/10630984926/job/29471519599) that seems unrelated, rerun helped.

**Review by nrcventura — APPROVED** [review]

**Pull request merged**